localstack/services/apigateway/resource_providers/aws_apigateway_deployment.py

```diff
@@ -0,0 +1,187 @@
+    DataTraceEnabled: Optional[bool]
+    Description: Optional[str]
+    DocumentationVersion: Optional[str]
+    LoggingLevel: Optional[str]
+    MethodSettings: Optional[list[MethodSetting]]
+    MetricsEnabled: Optional[bool]
+    Tags: Optional[list[Tag]]
+    ThrottlingBurstLimit: Optional[int]
+    ThrottlingRateLimit: Optional[float]
+    TracingEnabled: Optional[bool]
+    Variables: Optional[dict]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class ApiGatewayDeploymentProvider(ResourceProvider[ApiGatewayDeploymentProperties]):
+
+    TYPE = "AWS::ApiGateway::Deployment"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[ApiGatewayDeploymentProperties],
+    ) -> ProgressEvent[ApiGatewayDeploymentProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/DeploymentId
+          - /properties/RestApiId
+
+        Required properties:
+          - RestApiId
+
+        Create-only properties:
+          - /properties/DeploymentCanarySettings
+          - /properties/RestApiId
+
+        Read-only properties:
+          - /properties/DeploymentId
+
+        IAM permissions required:
+          - apigateway:POST
+
+        """
+        model = request.desired_state
+        api = request.aws_client_factory.apigateway
+
+        params = {"restApiId": model["RestApiId"]}
+
+        if model.get("StageName"):
+            params["stageName"] = model["StageName"]
+
+        if model.get("StageDescription"):
+            params["stageDescription"] = model["StageDescription"]
+
+        if model.get("Description"):
+            params["description"] = model["Description"]
+
+        response = api.create_deployment(**params)
+
+        model["DeploymentId"] = response["id"]
+
+        return ProgressEvent(
+            status=OperationStatus.SUCCESS,
+            resource_model=model,
+            custom_context=request.custom_context,
+        )
+
+    def read(
+        self,
+        request: ResourceRequest[ApiGatewayDeploymentProperties],
+    ) -> ProgressEvent[ApiGatewayDeploymentProperties]:
+        """
+        Fetch resource information
+
+        IAM permissions required:
+          - apigateway:GET
+        """
+        raise NotImplementedError
+
+    def delete(
+        self,
+        request: ResourceRequest[ApiGatewayDeploymentProperties],
+    ) -> ProgressEvent[ApiGatewayDeploymentProperties]:
+        """
+        Delete a resource
+
+        IAM permissions required:
+          - apigateway:GET
+          - apigateway:DELETE
+        """
+        model = request.desired_state
+        api = request.aws_client_factory.apigateway
+        try:
+            api.delete_deployment(restApiId=model["RestApiId"], deploymentId=model["DeploymentId"])
+        except api.exceptions.NotFoundException:
+            pass
```

Comment:
minor: maybe add a comment that this exception catching is because of a lack of proper dependency management in the CFn engine?

Reply:
Sounds good. I'll make the changes.